File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_03_二叉树修改与构造/LeetCode_0226.java
//226. 翻转二叉树
package com.kurumi.binarytree._03_二叉树修改与构造;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0226 {

}
class Solution {
  public TreeNode invertTree(TreeNode root) {
    if(root==null) return root;
    swap(root);
    invertTree(root.left);
    invertTree(root.right);
    return root;
  }
  public void swap(TreeNode root){
    TreeNode tmp=root.left;
    root.left=root.right;
    root.right=tmp;
  }
}

File: 卡码/代码随想录/1/06-二叉树/Go/03-二叉树修改与构造/LeetCode_0226.go
//226. 翻转二叉树
package main

import (
	. "binaryTree/types"
)

func invertTree(root *TreeNode) *TreeNode {
	if root==nil {
		return root
	}
	root.Left,root.Right=root.Right,root.Left
	invertTree(root.Left)
	invertTree(root.Right)
	return root
}
